quinn-proto/src/crypto/types.rs

```diff
@@ -80,9 +83,11 @@ impl std::iter::FromIterator<Certificate> for CertificateChain {
     }
 }
 
-impl From<Vec<rustls::Certificate>> for CertificateChain {
-    fn from(certs: Vec<rustls::Certificate>) -> Self {
-        Self { certs }
+impl From<&[rustls::Certificate]> for CertificateChain {
+    fn from(certs: &[rustls::Certificate]) -> Self {
+        Self {
+            certs: certs.to_vec(),
+        }
```

Comment:
Why this change? `From<Vec<_>>` is nice in that a user who already has a `Vec` can avoid some work, and others can call `to_vec` as needed.

Reply:
Fair enough, reverted. (I think I changed this to make it easier to feed directly from `TlsSession::peer_certificates()` into `CertificateChain`, but adding a `to_vec()` there seems fine.)